solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1365,6 +1382,32 @@ public void refreshAndWatch() {
         log.error("Unwatched collection: [{}]", coll, e);
       }
     }
+
+    private void refreshAndWatchChildren() throws KeeperException, InterruptedException {
+      Stat stat = new Stat();
+      List<String> replicaStates = null;
+      try {
+        replicaStates = zkClient.getChildren(collectionPath, this, stat, true);
+        PerReplicaStates newStates = new PerReplicaStates(collectionPath, stat.getCversion(), replicaStates);
+        DocCollection oldState = watchedCollectionStates.get(coll);
+        DocCollection newState = null;
```

Comment:
Please do not initialize `newState` to `null` and instead mark it `final` so compiler enforces it being assigned. (same for `replicaStates` above)